src/main/webapp/resources/js/apis/metadata/field.ts

```diff
@@ -5,9 +5,18 @@ import axios from "axios";
 import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 import { addKeysToList } from "../../utilities/http-utilities";
 import { setBaseUrl } from "../../utilities/url-utilities";
+import { Restriction } from "../../pages/projects/samples-metadata-import/redux/importReducer";
 
 const BASE_URL = setBaseUrl(`/ajax/metadata/fields`);
 
+export interface MetadataField {
+  id?: number;
```

Comment:
I'm not going to lie, I don't 100% understand what is considered a top level type. Initially, I thought it was only those extending the base model that were being moved to index.d.ts. Out of all the types in files under /webapp/resources/js/apis, I don't get which ones can stay and which ones are suppose to be moved into index.d.ts. For example, I don't know why it's okay for SequenceFile to be moved, but not FastQC. I've also kinda wondered how you're going to refrain from getting dupicating types, especially when they might not be exported. For example we have two interfaces called MetadataRoles, but it exists in EditMetadata.tsx and AddNewMetadata.tsx.
Regardless, please see 8e443022c311d3d118649cabcc2b50ad5b1a3948.

Reply:
You're right, a lot of them do need to be cleaned up (I am learning typescript as we go, just like you and @deepsidhu85 and @ericenns), and if you feel that you would like to take on that task please go ahead and do it.  My intention is that anything that exists as a core IRIDA model on the server (or returned as such) be represented in the `irida/index.d.ts` file so that these are standard definitions.  If the type you need does not match that one, you can make a new type in the file you are working on _based on the definition in that higher-level file_. 

Example.  Say in the root class that the MetadataField id was not optional, but since you are creating new ones you want that to be.  In your type you:

```typescript
type NewMetadataField = Partial<MetadataField>;
```

or

```typescript
type NewMetadataField = Omit<MetadataFeild, "id">
```

Please remember that we are all trying to figure this out.